### PR TITLE
Fix type template dropdown and form issues

### LINKED ISSUE #112: Default state `Choose a type template` shown in `Set Schema` dropdown, even if there is template selected
image

### PR DESCRIPTION
**Pull request by Datron**
## Problem
This PR closes #112 and #111 

## Solution
Screenshot 2024-06-20 at 4 50 23 PM
Screenshot 2024-06-20 at 4 50 35 PM

### CODE REVIEW

File: crates/frontend/src/components/default_config_form.rs [hunk 1]
```diff
@@ -150,12 +150,18 @@ where
             <Suspense>
                 {move || {
                     let options = type_template_resource.get().unwrap_or(vec![]);
-                    let config_t = if config_type_rs.get().is_empty() {
+                    let config_t = if config_type_rs.get().is_empty() && edit {
+                        "change current type template".into()
```
Comment (sauraww) [inline crates/frontend/src/components/default_config_form.rs:154]: Why are we allowing the default config type to change ?
Wouldn't that be an issue for existing overrides , just wanted to understand the use case where we would need it.

**Review by sauraww — CHANGES_REQUESTED** [review]

File: crates/frontend/src/components/override_form.rs [hunk 2]
```diff
@@ -117,7 +117,7 @@ where
                                                             .map(ConfigType::DefaultConfig)
                                                             .collect::<Vec<_>>(),
                                                     )
-                                                    .expect("can't parse default config key");
+                                                    .unwrap_or(json!(input_value));
```
Comment (sauraww) [inline crates/frontend/src/components/override_form.rs:120]: @Datron we would also need to make the similar changes at line number 35 and 51 in `crates/frontend/src/components/context_form/utils.rs `

**Review by sauraww — CHANGES_REQUESTED** [review]

File: crates/frontend/src/components/default_config_form.rs (hunk 1 above)
Comment (Datron) [reply crates/frontend/src/components/default_config_form.rs:154]: @sauraww adding an enum, changing the pattern - these are few examples where we would want to update the type. I replicated what was the behaviour today. If we want to stop this, we discuss and do it in another PR

File: crates/frontend/src/components/override_form.rs (hunk 2 above)
Comment (Datron) [reply crates/frontend/src/components/override_form.rs:120]: @sauraww why? There is no type templates in context form

Comment (Datron) [reply crates/frontend/src/components/override_form.rs:120]: If this is an issue, it seems out of scope of this PR

File: crates/frontend/src/components/default_config_form.rs (hunk 1 above)
Comment (mahatoankitkumar) [reply crates/frontend/src/components/default_config_form.rs:154]: @sauraww this functionality of changing the schema of a default_config is already there. And the change would effect any new overrides i think, it wouldn't effect the existing ones i guess.

**Review by mahatoankitkumar — APPROVED** [review]

**Pull request merged**